channels/api.py

```diff
@@ -361,3 +363,52 @@ def more_comments(self, comment_fullname, parent_fullname, count, children=None)
         more_comments.submission = self.reddit.submission(submission_id)
         more_comments.comments()  # load the comments
         return more_comments
+
+    def is_subscribe(self, channel_name):
+        is_subscribed = False
+        list = self.list_channels()
+        for reddit in list:
+            if reddit.display_name == channel_name:
+                is_subscribed = True
+                break
+        return is_subscribed
+
+    def subscribe(self, channel_name):
+        """
+        Subscribe a user to a channel
+
+        Args:
+            channel_name(str): the channel name identifier
+
+        Returns:
+            channels.models.UnSubscribeStatus: the subscription status of user
+        """
+        status, __ = UnSubscribeStatus.objects.get_or_create(user=self.user, channel_name=channel_name)
+        if not self.is_subscribe(channel_name):
+            self.get_channel(channel_name).subscribe()
+            status.is_unsubscribe = False
+            status.save()
+
+        return {
+            'is_subscribed': self.is_subscribe(channel_name)
+        }
+
+    def unsubscribe(self, channel_name):
+        """
+        Unsubscribe a user to a channel
+
+        Args:
+            channel_name(str): the channel name identifier
+
+        Returns:
+            channels.models.UnSubscribeStatus: the subscription status of user
+        """
+        status, __ = UnSubscribeStatus.objects.get_or_create(user=self.user, channel_name=channel_name)
```

Comment:
why we are tracking un subscription of users, if you unsubscribe what will happen then